lib/galaxy/model/__init__.py

```diff
@@ -923,6 +923,40 @@ def adjust_total_disk_usage(self, amount, quota_source_label):
                 }
                 sa_session.execute(statement, params)
 
+    def _get_social_auth(self, provider_backend):
+        if not self.social_auth:
+            return None
+        for auth in self.social_auth:
+            if auth.provider == provider_backend and auth.extra_data:
+                return auth
+        return None
+
+    def _get_custos_auth(self, provider_backend):
+        if not self.custos_auth:
+            return None
+        for auth in self.custos_auth:
+            if auth.provider == provider_backend and auth.refresh_token:
+                return auth
+        return None
+
+    def get_oidc_tokens(self, provider_backend):
```

Comment:
Should this be `get_oidc_token` since you're just getting a single token ?

Reply:
I return a dictionary with three tokens - _access_, _refresh_ and _id_